{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 0: Import\n",
    "**Import the various libraries**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 1: Read Files\n",
    "**Raw data:**\n",
    "**(1) cycle 1 & cycle 35**\n",
    "**(2) genotype**\n",
    "**(3) risk factor**\n",
    "**(4) disease risk**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Disease_code   rs_number\n",
      "85          D12  rs10865035\n",
      "86          D12   rs2230926\n",
      "87          D12   rs9888739\n",
      "88          D13   rs1024161\n",
      "89          D13    rs179247\n",
      "90          D13  rs12101261\n",
      "91          D13   rs1800630\n",
      "92          D13   rs2228570\n",
      "93          D15      rs7412\n",
      "94          D15    rs429358\n",
      "95          D15   rs7101429\n",
      "96          D15    rs610932\n",
      "97          D15  rs10845990\n",
      "98          D15    rs334558\n",
      "99          D16    rs242562\n"
     ]
    }
   ],
   "source": [
    "################################\n",
    "# Read files: cycle 1 & cycle 35\n",
    "################################\n",
    "\n",
    "cycle1 = pd.read_excel(open('./raw/G2190712-01c-cycle 1.xls', 'rb'),\n",
    "              sheet_name='Multicomponent Data')\n",
    "\n",
    "\n",
    "cycle35 = pd.read_excel(open('./raw/G2190712-01c-cycle 35.xls', 'rb'),\n",
    "              sheet_name='Multicomponent Data')\n",
    "\n",
    "\n",
    "cycle1 = pd.DataFrame(cycle1)\n",
    "cycle1 = cycle1.loc[46:]\n",
    "cycle1.columns = ['Well', 'Cycle', 'ROX', 'VIC', 'FAM']\n",
    "#cycle1.head()\n",
    "\n",
    "cycle35 = pd.DataFrame(cycle35)\n",
    "cycle35 = cycle35.loc[46:]\n",
    "cycle35.columns = ['Well', 'Cycle', 'ROX', 'VIC', 'FAM']\n",
    "#cycle35.head()\n",
    "\n",
    "\n",
    "\n",
    "##########################\n",
    "# Read files: genotype.csv\n",
    "##########################\n",
    "\n",
    "\n",
    "genotype = pd.read_csv('./raw/genotype.csv')\n",
    "\n",
    "#print(genotype.head())\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#############################\n",
    "# Read files: risk_factor.csv\n",
    "#############################\n",
    "\n",
    "\n",
    "risk_factor = pd.read_csv('./raw/risk_factor.csv')\n",
    "\n",
    "#risk_factor.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############################\n",
    "# Read files: disease_risk.csv\n",
    "##############################\n",
    "\n",
    "disease_risk = pd.read_csv('./raw/disease_risk.csv')\n",
    "\n",
    "print(disease_risk.iloc[85:100])\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##########################\n",
    "# Read files: nutrient.csv\n",
    "##########################\n",
    "\n",
    "\n",
    "nutrient = pd.read_csv('./raw/nutrient.csv', encoding = 'utf-8')\n",
    "\n",
    "#nutrient.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###########################\n",
    "# Read files: lifestyle.csv\n",
    "###########################\n",
    "\n",
    "\n",
    "lifestyle = pd.read_csv('./raw/lifestyle.csv', encoding = 'utf-8')\n",
    "\n",
    "#lifestyle.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#############################\n",
    "# Read files: risk_factor.csv\n",
    "#############################\n",
    "\n",
    "\n",
    "risk_factor = pd.read_csv('./raw/risk_factor.csv', encoding = 'utf-8')\n",
    "\n",
    "#risk_factor.head()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11\n",
      "['GT', 'AA', 'AG', 'CC', 'GG', 'AA', 'AG', 'CT', 'TT', 'GG', 'TT', 'TT', 'GG', 'TT', 'CC', 'GG', 'AA', 'GG', 'CT', 'CT', 'TT', 'GG', 'TT', 'AG', 'TT', 'AT', 'CT', 'GG', 'CT', 'CC', 'GG', 'TT', 'AG', 'GA', 'CT', 'CC', 'CG', 'CT', 'CC', 'CC', 'AG', 'CC', 'CT', 'AG', 'CT', 'GG', 'GT', 'AG', 'CT', 'AG', 'TT', 'AG', 'TT', 'AG', 'CC', 'AA', 'AG', 'GG', 'AG', 'AG', 'CG', 'GT', 'TT', 'CC', 'TT', 'AG', 'AA', 'CT', 'CC', 'GA', 'GA', 'GG', 'GG', 'GG', 'TT', 'AA', 'TT', 'CT', 'CC', 'CC', 'CC', 'GG', 'TT', 'CC', 'GG', 'TT', 'AG', 'GG', 'GG', 'TT', 'CT', 'CT', 'CC', 'AG', 'CT', 'GG', 'CC', 'TT', 'GG', 'CC', 'CC', 'TG', 'CC', 'AC', 'TT', 'AA', 'CC', 'GG', 'GG', 'CT', 'GG', 'AG', 'TT', 'CT', 'TT', 'TT', 'TT', 'CT', 'CC', 'CC', 'AG', 'AG', 'GG', 'GG', 'TT', 'AC', 'GG', 'AA', 'GG', 'CC', 'TT', 'AG', 'AA', 'CC', 'CT', 'GA', 'GG', 'AG', 'CC', 'TT', 'CC', 'TT', 'CT', 'AA', 'CT', 'AG', 'TT', 'CT', 'AA', 'GG', 'CT', 'TT', 'GG', 'CT', 'AA', 'CT', 'GT', 'CC', 'GG', 'CC', 'CT', 'CT', 'TT', 'CT', 'TT', 'AC', 'TT', 'GG', 'CG', 'GT', 'TT', 'AC', 'CC', 'GG', 'TT', 'AA', 'CT', 'AA', 'TT', 'CC', 'GG', 'AG', 'TT', 'CC', 'TT', 'AG', 'AA', 'AA', 'TT', 'GG', 'CC', 'AC', 'TT', 'AA', 'TT', 'GG', 'CA', 'CC', 'AA', 'TT', 'GG', 'AG', 'AG', 'GG', 'GG', 'AA', 'AA', 'TT', 'CT', 'CC', 'AG', 'AA', 'AA', 'CC', 'AG', 'GG', 'CT', 'CT', 'CC', 'GG', 'CG', 'GG', 'CT', 'TT', 'TT', 'AG', 'GG', 'GG', 'TT', 'AG', 'GT', 'GT', 'CC', 'AG', 'AA', 'TT', 'CC', 'CC', 'GG', 'AA', 'AG', 'GT', 'CG', 'TT', 'GG', 'TT', 'CC', 'TT', 'GG', 'AA', 'GG', 'CC', 'CT', 'CC', 'GG', 'GG', 'TT', 'GG', 'GG', 'TT', 'AG', 'AC', 'GT', 'TG', 'CC', 'GG', 'CT', 'GT', 'TT', 'TT', 'GA', 'GT', 'CT', 'AA', 'CT', 'AA', 'TT', 'AG', 'CC', 'CC', 'CC', 'TT', 'AG', 'CG', 'GG', 'GG', 'TT', 'CT', 'AG', 'CT', 'TT', 'GT', 'GG', 'GG', 'AA', 'TT', 'CG', 'CC', 'GG', 'GG', 'TT', 'GG', 'CC', 'GG', 'GG', 'CC', 'AG', 'AT', 'TT', 'CC', 'CT', 'CC', 'AC', 'TT', 'CT', 'AC', 'CT', 'CC', 'CC', 'GG', 'AA', 'TT', 'GT', 'CT', 'CT', 'AT', 'GT', 'AA', 'GG', 'CC', 'AA', 'TT', 'AA', 'CT', 'CC', 'GG', 'CC', 'TT', 'CC', 'CT', 'CC', 'GG', 'GG', 'GG', 'AA', 'AG', 'AG', 'CT', 'CT', 'AG', 'CT', 'GG', 'GG', 'GG', 'TT', 'TG', 'CC', 'AA', 'TT', 'CT', 'CT', 'AT', 'GG', 'CC', 'CT', 'CT', 'TT', 'TT', 'CC', 'AG', 'GT', 'GG', 'TT', 'CT', 'CC', 'TT', 'GG', 'CT', 'CC', 'GG', 'GG', 'AA']\n",
      "468 468 468\n"
     ]
    }
   ],
   "source": [
    "#############################\n",
    "#1. Genotype: delta VIC & FAM \n",
    "#############################\n",
    "\n",
    "\n",
    "\n",
    "#################################\n",
    "# calculate delta_vic & delta_fam\n",
    "\n",
    "delta_vic = list(cycle35['VIC'] - cycle1['VIC'])\n",
    "#print(delta_vic)\n",
    "\n",
    "delta_fam = list(cycle35['FAM'] - cycle1['FAM'])\n",
    "#print(delta_fam)\n",
    "\n",
    "delta = pd.DataFrame(list(zip(delta_vic, delta_fam)), \n",
    "                    columns=['delta_vic','delta_fam'])\n",
    "\n",
    "    \n",
    "    \n",
    "delta.head()\n",
    "\n",
    "\n",
    "\n",
    "#############################\n",
    "# get the signal of delta_vic\n",
    "\n",
    "vic_signal = []\n",
    "\n",
    "for cell in delta_vic:\n",
    "    if cell >= 500:\n",
    "        vic_signal.append(1)\n",
    "        \n",
    "    else:\n",
    "        vic_signal.append(0)\n",
    "    \n",
    "#print(vic_signal)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#############################\n",
    "# get the signal of delta_fam\n",
    "\n",
    "fam_signal = []\n",
    "\n",
    "for cell in delta_fam:\n",
    "    if cell >= 500:\n",
    "        fam_signal.append(1)\n",
    "        \n",
    "    else:\n",
    "        fam_signal.append(0)\n",
    "    \n",
    "#print(fam_signal)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "########################################\n",
    "# get the error of delta_vic & delta_fam\n",
    "\n",
    "signal = []\n",
    "\n",
    "for i in range(0,len(vic_signal)):\n",
    "    signal.append(vic_signal[i] + fam_signal[i])\n",
    "\n",
    "#print(signal)\n",
    "\n",
    "\n",
    "error = 0\n",
    "for cell in signal:\n",
    "    if cell < 2:\n",
    "        error += 1\n",
    "\n",
    "print(error)\n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "    \n",
    "######################################\n",
    "# get the value: delta_vic / delta_fam\n",
    "\n",
    "value = []\n",
    "\n",
    "for i in range(len(delta_vic )):\n",
    "    \n",
    "    if vic_signal[i] == 1 and fam_signal[i] == 1:\n",
    "        \n",
    "        value.append(delta_vic[i] / delta_fam[i])\n",
    "    \n",
    "    elif vic_signal[i] != 1 and fam_signal[i] == 1:\n",
    "        \n",
    "        value.append(1 / delta_fam[i])\n",
    "\n",
    "    elif vic_signal[i] == 1 and fam_signal[i] != 1:\n",
    "        \n",
    "        value.append(delta_vic[i] / 1)\n",
    "        \n",
    "    else:\n",
    "        value.append(1)\n",
    "\n",
    "#print(value)\n",
    "\n",
    "\n",
    "\n",
    "#########################################\n",
    "# get the genotype: delta_vic / delta_fam\n",
    "\n",
    "\n",
    "\n",
    "client_genotype = []\n",
    "\n",
    "\n",
    "for i in range(len(value)-2):\n",
    "                  #standard1\n",
    "    if value[i] > genotype.iloc[i, 9]:\n",
    "        \n",
    "        #allele1\n",
    "        client_genotype.append(genotype.iloc[i, 6])\n",
    "                    #standard2\n",
    "    elif value[i] < genotype.iloc[i, 10]:\n",
    "        \n",
    "        #allele3\n",
    "        client_genotype.append(genotype.iloc[i, 8])\n",
    "        \n",
    "    else:\n",
    "        \n",
    "        #allele2\n",
    "        client_genotype.append(genotype.iloc[i, 7])\n",
    "\n",
    "\n",
    "    \n",
    "#print(client_genotype)\n",
    "\n",
    "print(client_genotype)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##########################################\n",
    "# get the risk factor of specific genotype\n",
    "##########################################\n",
    "\n",
    "\n",
    "disease_risk.head()\n",
    "\n",
    "\n",
    "# match the genotype to disease_risk\n",
    "rs_genetic_risk = []\n",
    "\n",
    "for rs_disease in disease_risk['rs_number']:\n",
    "    \n",
    "    for j in range(len(genotype['rs number'])):\n",
    "    \n",
    "        if rs_disease == genotype['rs number'][j]:\n",
    "            \n",
    "            rs_genetic_risk.append(client_genotype[j])\n",
    "\n",
    "\n",
    "         \n",
    "            \n",
    "disease_genotype = pd.DataFrame(list(zip(disease_risk['Disease_code'], disease_risk['rs_number'], rs_genetic_risk)), columns=['Disease_code', 'rs_number', 'genotype'])            \n",
    "   \n",
    "#df.to_csv('./raw/rs_genetic_risk.csv')  \n",
    "    \n",
    "\n",
    "print(len(disease_risk['Disease_code']), len(disease_risk['rs_number']), len(rs_genetic_risk) )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "468\n"
     ]
    }
   ],
   "source": [
    "##########################################\n",
    "# get the risk factor of specific genotype\n",
    "##########################################\n",
    "\n",
    "\n",
    "disease_risk.head()\n",
    "\n",
    "\n",
    "# match the genotype to disease_risk\n",
    "\n",
    "rs_genetic_risk = []\n",
    "\n",
    "for rs_disease in disease_risk['rs_number']:\n",
    "    \n",
    "    for j in range(len(genotype['rs number'])):\n",
    "    \n",
    "        if rs_disease == genotype['rs number'][j]:\n",
    "            \n",
    "            rs_genetic_risk.append(client_genotype[j])\n",
    "\n",
    "\n",
    "\n",
    "            \n",
    "# get the encode of each allele risk\n",
    "\n",
    "risk_encode = []\n",
    "\n",
    "for i in range(len(disease_risk['Disease_code'])):\n",
    "    \n",
    "    encode = (disease_risk['Disease_code'][i] + disease_risk['rs_number'][i] + rs_genetic_risk[i])\n",
    "    \n",
    "    risk_encode.append(encode)\n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "            \n",
    "disease_genotype = pd.DataFrame(list(zip(disease_risk['Disease_code'], disease_risk['rs_number'], rs_genetic_risk, risk_encode)), columns=['Disease_code', 'rs_number', 'genotype', 'risk_encode'])            \n",
    "   \n",
    "disease_genotype.head()    \n",
    "    \n",
    "    \n",
    "    \n",
    "#df.to_csv('./raw/disease_genotype.csv')\n",
    "\n",
    "print(len(risk_encode))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3601 3601 0\n"
     ]
    }
   ],
   "source": [
    "a = list(risk_factor['Allele_risk'])\n",
    "\n",
    "len(a)\n",
    "\n",
    "unique_risk = list(set(a))\n",
    "\n",
    "len(unique_risk)\n",
    "\n",
    "\n",
    "\n",
    "b = set([x for x in a if a.count(x) > 1])\n",
    "print(len(a), len(unique_risk), len(b))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "468"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "risk = []       \n",
    "        \n",
    "for encode in risk_encode:\n",
    "    \n",
    "    if encode in list(risk_factor['Allele_risk']):\n",
    "        \n",
    "        a = risk_factor.loc[risk_factor['Allele_risk'] == encode].index[0]\n",
    "    \n",
    "        risk.append(risk_factor.iloc[a,1])\n",
    "        \n",
    "    elif encode not in list(risk_factor['Allele_risk']):\n",
    "        \n",
    "        risk.append(1)\n",
    "        \n",
    "\n",
    "\n",
    "         \n",
    " \n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "gene_risk = pd.DataFrame(list(zip(disease_risk['Disease_code'], disease_risk['rs_number'], rs_genetic_risk, risk_encode, risk)), columns=['Disease_code', 'rs_number', 'genotype', 'risk_encode', 'risk'])             \n",
    "\n",
    "\n",
    "gene_risk.head()\n",
    "\n",
    "gene_risk.to_csv('./raw/gene_risk.csv')\n",
    "\n",
    "\n",
    "len(risk)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.043\n"
     ]
    }
   ],
   "source": [
    "risk = []\n",
    "\n",
    "        \n",
    "        \n",
    "        \n",
    "for encode in risk_encode:\n",
    "    \n",
    "    if encode in list(risk_factor['Allele_risk']):\n",
    "        \n",
    "        a = risk_factor.loc[risk_factor['Allele_risk'] == encode].index[0]\n",
    "    \n",
    "        risk.append(risk_factor.iloc[a,1])\n",
    "        \n",
    "    elif encode not in list(risk_factor['Allele_risk']):\n",
    "        \n",
    "        risk.append(1)\n",
    "    \n",
    "\n",
    "len(risk)\n",
    "print(risk[99])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "# test  cell\n",
    "#disease_genotype.loc[disease_genotype['risk_encode'] == 'D16rs242562AG'].index[0]\n",
    "\n",
    "\n",
    "#risk_factor['Allele_risk'] == 'D16rs242562AG'\n",
    "\n",
    "#a = risk_factor.loc[risk_factor['Allele_risk'] == 'D16rs242562A'].index[0]\n",
    "#print(a)\n",
    "\n",
    "if 'D16rs242562AG' in list(risk_factor['Allele_risk']):\n",
    "    print('true')\n",
    "\n",
    "elif 'D16rs242562AG' not in list(risk_factor['Allele_risk']):\n",
    "    print('no')\n",
    "   \n",
    "#type(a)\n",
    "#risk_factor.iloc[a,1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.9087'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "risk[98]\n",
    "\n",
    "#gene_risk.iloc[0:12,4]\n",
    "\n",
    "#print(d1_risk[0], d1_risk[3], d1_risk[5], d1_risk[7], d1_risk[8])\n",
    "#max(d1_risk[0], d1_risk[3], d1_risk[5], d1_risk[7], d1_risk[8])\n",
    "#max(d1_risk[2], d1_risk[4], d1_risk[11])\n",
    "#d1_risk[9]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.63 1.16\n"
     ]
    }
   ],
   "source": [
    "##############\n",
    "# Disease Risk\n",
    "##############\n",
    "\n",
    "risk = pd.to_numeric(risk, downcast='float')\n",
    "\n",
    "\n",
    "###########\n",
    "# D1: 肥胖症\n",
    "\n",
    "d1 = gene_risk.iloc[0:12]\n",
    "\n",
    "\n",
    "d1_g1 = max(risk[0], risk[3], risk[5], risk[7], risk[8])\n",
    "d1_g2 = risk[1]\n",
    "d1_g3 = max(risk[2], risk[4], risk[11])\n",
    "d1_g4 = risk[6]\n",
    "d1_g5 = risk[9]\n",
    "\n",
    "d1_ratio = (d1_g1*d1_g2*d1_g3*d1_g4*d1_g5)*1.2\n",
    "\n",
    "d1_ratio = format(d1_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# D2: 高尿酸血症\n",
    "\n",
    "d2 = gene_risk.iloc[12:20]\n",
    "\n",
    "\n",
    "d2_g1 = max(risk[12], risk[13], risk[21])\n",
    "d2_g2 = max(risk[14], risk[15])\n",
    "d2_g3 = risk[16]\n",
    "d2_g4 = max(risk[17], risk[18])\n",
    "\n",
    "\n",
    "\n",
    "d2_ratio = (d2_g1*d2_g2*d2_g3*d2_g4)\n",
    "\n",
    "d2_ratio = format(d2_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################\n",
    "# D3: 第二型糖尿病\n",
    "\n",
    "d3 = gene_risk.iloc[20:30]\n",
    "\n",
    "\n",
    "d3_g1 = max(risk[20], risk[21], risk[25], risk[29])\n",
    "d3_g2 = max(risk[22], risk[23])\n",
    "d3_g3 = risk[24]\n",
    "d3_g4 = risk[26]\n",
    "d3_g5 = risk[27]\n",
    "d3_g6 = risk[28]\n",
    "\n",
    "\n",
    "\n",
    "d3_ratio = (d3_g1*d3_g2*d3_g3*d3_g4*d3_g5*d3_g6) * 1.2\n",
    "\n",
    "d3_ratio = format(d3_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "################\n",
    "# D4: 第二型糖尿病\n",
    "\n",
    "d4 = gene_risk.iloc[20:30]\n",
    "\n",
    "\n",
    "d4_g1 = risk[30]\n",
    "d4_g2 = max(risk[31], risk[32], risk[33])\n",
    "d4_g3 = max(risk[34], risk[37], risk[38], risk[39])\n",
    "d4_g4 = max(risk[35], risk[36])\n",
    "\n",
    "\n",
    "\n",
    "d4_ratio = (d4_g1*d4_g2*d4_g3*d4_g4)\n",
    "\n",
    "d4_ratio = format(d4_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###########\n",
    "# D5: 腦中風\n",
    "\n",
    "d5 = gene_risk.iloc[40:44]\n",
    "\n",
    "\n",
    "d5_g1 = risk[40]\n",
    "d5_g2 = risk[41]\n",
    "d5_g3 = risk[42]\n",
    "d5_g4 = risk[43]\n",
    "\n",
    "\n",
    "\n",
    "d5_ratio = (d5_g1*d5_g2*d5_g3*d5_g4)*1.2\n",
    "\n",
    "d5_ratio = format(d5_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###########\n",
    "# D6: 腦中風\n",
    "\n",
    "d6 = gene_risk.iloc[40:44]\n",
    "\n",
    "\n",
    "d6_g1 = risk[44]\n",
    "d6_g2 = risk[45]\n",
    "d6_g3 = risk[46]\n",
    "d6_g4 = risk[47]\n",
    "d6_g5 = risk[48]\n",
    "\n",
    "\n",
    "d6_ratio = (d6_g1*d6_g2*d6_g3*d6_g4*d6_g5)*1.2\n",
    "\n",
    "d6_ratio = format(d6_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###########\n",
    "# D7: 高血壓\n",
    "\n",
    "d7 = gene_risk.iloc[49:58]\n",
    "\n",
    "\n",
    "d7_g1 = max(risk[49], risk[50], risk[56])\n",
    "d7_g2 = risk[51]\n",
    "d7_g3 = risk[52]\n",
    "d7_g4 = max(risk[53], risk[54])\n",
    "d7_g5 = risk[55]\n",
    "d7_g6 = risk[57]\n",
    "\n",
    "\n",
    "d7_ratio = (d7_g1*d7_g2*d7_g3*d7_g4*d7_g5*d7_g6)\n",
    "\n",
    "d7_ratio = format(d7_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###########\n",
    "# D8: 高血壓\n",
    "\n",
    "d8 = gene_risk.iloc[58:68]\n",
    "\n",
    "\n",
    "d8_g1 = max(risk[58], risk[61], risk[63], risk[66])\n",
    "d8_g2 = risk[59]\n",
    "d8_g3 = risk[60]\n",
    "d8_g4 = risk[62]\n",
    "d8_g5 = risk[64]\n",
    "d8_g6 = risk[65]\n",
    "d8_g7 = risk[67]\n",
    "\n",
    "d8_ratio = (d8_g1*d8_g2*d8_g3*d8_g4*d8_g5*d8_g6*d8_g7)\n",
    "\n",
    "d8_ratio = format(d8_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "#################\n",
    "# D10: 僵直性脊椎炎\n",
    "\n",
    "d10 = gene_risk.iloc[68:72]\n",
    "\n",
    "\n",
    "d10_g1 = risk[68]\n",
    "d10_g2 = risk[69]\n",
    "d10_g3 = risk[70]\n",
    "d10_g4 = risk[71]\n",
    "\n",
    "\n",
    "d10_ratio = (d10_g1*d10_g2*d10_g3*d10_g4)\n",
    "\n",
    "d10_ratio = format(d10_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "###################\n",
    "# D11: 類風濕性關節炎\n",
    "\n",
    "d11 = gene_risk.iloc[72:80]\n",
    "\n",
    "\n",
    "d11_g1 = max(risk[72], risk[73])\n",
    "d11_g2 = max(risk[74], risk[75])\n",
    "d11_g3 = max(risk[76], risk[77])\n",
    "d11_g4 = risk[78]\n",
    "d11_g5 = risk[79]\n",
    "\n",
    "\n",
    "d11_ratio = (d11_g1*d11_g2*d11_g3*d11_g4*d11_g5)*0.95\n",
    "\n",
    "d11_ratio = format(d11_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###############\n",
    "# D12: 紅斑性狼瘡\n",
    "\n",
    "\n",
    "d12 = gene_risk.iloc[80:88]\n",
    "\n",
    "\n",
    "d12_g1 = risk[81]\n",
    "d12_g2 = risk[82]\n",
    "d12_g3 = max(risk[80], risk[83], risk[87])\n",
    "\n",
    "\n",
    "d12_ratio = (d12_g1*d12_g2*d12_g3)\n",
    "\n",
    "d12_ratio = format(d12_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "######################\n",
    "# D13: 自體免疫甲狀腺疾病\n",
    "\n",
    "\n",
    "d13 = gene_risk.iloc[88:93]\n",
    "\n",
    "\n",
    "d13_g1 = max(risk[88], risk[91], risk[92])\n",
    "d13_g2 = max(risk[89], risk[90])\n",
    "\n",
    "\n",
    "\n",
    "d13_ratio = (d13_g1*d13_g2)\n",
    "\n",
    "d13_ratio = format(d13_ratio, '.2f')\n",
    "\n",
    "\n",
    "###############\n",
    "# D15: 阿茲海默症\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###############\n",
    "# D16: 帕金森氏症\n",
    "\n",
    "\n",
    "d16 = gene_risk.iloc[99:108]\n",
    "\n",
    "\n",
    "d16_g1 = max(risk[99], risk[105], risk[106], risk[107])\n",
    "d16_g2 = risk[100]\n",
    "d16_g3 = risk[101]\n",
    "d16_g4 = risk[102]\n",
    "d16_g5 = max(risk[103], risk[104])\n",
    "\n",
    "\n",
    "d16_ratio = (d16_g1*d16_g2*d16_g3*d16_g4*d16_g5)\n",
    "\n",
    "d16_ratio = format(d16_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# D17: 骨質疏鬆\n",
    "\n",
    "\n",
    "d17 = gene_risk.iloc[108:115]\n",
    "\n",
    "\n",
    "d17_g1 = risk[112]\n",
    "d17_g2 = risk[113]\n",
    "d17_g3 = risk[114]\n",
    "\n",
    "\n",
    "d17_ratio = (d17_g1*d17_g2*d17_g3)*1.1\n",
    "\n",
    "d17_ratio = format(d17_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#################\n",
    "# D18: 退化性關節炎\n",
    "\n",
    "\n",
    "d18 = gene_risk.iloc[108:115]\n",
    "\n",
    "\n",
    "d18_g1 = max(risk[115], risk[121], risk[122], risk[123])\n",
    "d18_g2 = risk[116]\n",
    "d18_g3 = risk[117]\n",
    "d18_g4 = max(risk[118], risk[120])\n",
    "d18_g5 = risk[119]\n",
    "\n",
    "\n",
    "d18_ratio = (d18_g1*d18_g2*d18_g3*d18_g4*d18_g5)\n",
    "\n",
    "d18_ratio = format(d18_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D19: 白內障\n",
    "\n",
    "\n",
    "d19 = gene_risk.iloc[124:128]\n",
    "\n",
    "\n",
    "d19_g1 = risk[124]\n",
    "d19_g2 = risk[125]\n",
    "d19_g3 = risk[126]\n",
    "d19_g4 = risk[127]\n",
    "\n",
    "\n",
    "\n",
    "d19_ratio = (d19_g1*d19_g2*d19_g3*d19_g4)\n",
    "\n",
    "d19_ratio = format(d19_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "################\n",
    "# D20: 黃斑部病變\n",
    "\n",
    "\n",
    "d20 = gene_risk.iloc[128:135]\n",
    "\n",
    "\n",
    "d20_g1 = max(risk[128], risk[133], risk[131])\n",
    "d20_g2 = max(risk[129], risk[130])\n",
    "d20_g3 = risk[132]\n",
    "d20_g4 = risk[134]\n",
    "\n",
    "\n",
    "\n",
    "d20_ratio = (d20_g1*d20_g2*d20_g3*d20_g4)\n",
    "\n",
    "d20_ratio = format(d20_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D21: 青光眼\n",
    "\n",
    "\n",
    "d21 = gene_risk.iloc[128:135]\n",
    "\n",
    "\n",
    "d21_g1 = max(risk[135], risk[138], risk[140])\n",
    "d21_g2 = risk[136]\n",
    "d21_g3 = risk[137]\n",
    "d21_g4 = risk[139]\n",
    "\n",
    "\n",
    "\n",
    "d21_ratio = (d21_g1*d21_g2*d21_g3*d21_g4)\n",
    "d21_ratio = format(d21_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "###################\n",
    "# D22: 慢性阻塞性肺病\n",
    "\n",
    "\n",
    "d22 = gene_risk.iloc[141:146]\n",
    "\n",
    "\n",
    "d22_g1 = risk[141]\n",
    "d22_g2 = risk[142]\n",
    "d22_g3 = risk[143]\n",
    "d22_g4 = risk[144]\n",
    "d22_g5 = risk[145]\n",
    "\n",
    "\n",
    "\n",
    "d22_ratio = (d22_g1*d22_g2*d22_g3*d22_g4*d22_g5)\n",
    "d22_ratio = format(d22_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D23: 肝硬化\n",
    "\n",
    "\n",
    "d23 = gene_risk.iloc[146:155]\n",
    "\n",
    "\n",
    "d23_g1 = max(risk[146], risk[147], risk[149], risk[153])\n",
    "d23_g2 = risk[148]\n",
    "d23_g3 = risk[150]\n",
    "d23_g4 = risk[151]\n",
    "d23_g5 = risk[152]\n",
    "d23_g6 = risk[154]\n",
    "\n",
    "\n",
    "d23_ratio = (d23_g1*d23_g2*d23_g3*d23_g4*d23_g5*d23_g6)\n",
    "d23_ratio = format(d23_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D24: 脂肪肝\n",
    "\n",
    "\n",
    "d24 = gene_risk.iloc[155:164]\n",
    "\n",
    "\n",
    "d24_g1 = max(risk[155], risk[162])\n",
    "d24_g2 = max(risk[156], risk[157])\n",
    "d24_g3 = risk[158]\n",
    "d24_g4 = max(risk[159], risk[160], risk[161], risk[163])\n",
    "d24_g5 = risk[163]\n",
    "\n",
    "\n",
    "d24_ratio = (d24_g1*d24_g2*d24_g3*d24_g4*d24_g5)\n",
    "d24_ratio = format(d24_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D25: 膽結石\n",
    "\n",
    "\n",
    "d25 = gene_risk.iloc[164:173]\n",
    "\n",
    "\n",
    "d25_g1 = risk[164]\n",
    "d25_g2 = risk[165]\n",
    "d25_g3 = risk[166]\n",
    "d25_g4 = max(risk[167], risk[168], risk[169], risk[170])\n",
    "d25_g5 = risk[171]\n",
    "d25_g6 = risk[172]\n",
    "\n",
    "\n",
    "d25_ratio = (d25_g1*d25_g2*d25_g3*d25_g4*d25_g5*d25_g6)\n",
    "d25_ratio = format(d25_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D26: 腎結石\n",
    "\n",
    "\n",
    "d26 = gene_risk.iloc[173:179]\n",
    "\n",
    "\n",
    "d26_g1 = risk[173]\n",
    "d26_g2 = max(risk[174], risk[175])\n",
    "d26_g3 = risk[176]\n",
    "d26_g4 = risk[177]\n",
    "d26_g5 = risk[178]\n",
    "\n",
    "\n",
    "d26_ratio = (d26_g1*d26_g2*d26_g3*d26_g4*d26_g5)\n",
    "d26_ratio = format(d26_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##########################\n",
    "# D27: 消化性潰瘍、胃食道逆流\n",
    "\n",
    "\n",
    "d27 = gene_risk.iloc[179:185]\n",
    "\n",
    "\n",
    "d27_g1 = risk[179]\n",
    "d27_g2 = risk[180]\n",
    "d27_g3 = max(risk[181], risk[182])\n",
    "d27_g4 = risk[183]\n",
    "d27_g5 = risk[184]\n",
    "\n",
    "\n",
    "d27_ratio = (d27_g1*d27_g2*d27_g3*d27_g4*d27_g5)\n",
    "d27_ratio = format(d27_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################\n",
    "# D28: 大腸息肉症\n",
    "\n",
    "\n",
    "d28 = gene_risk.iloc[185:192]\n",
    "\n",
    "\n",
    "d28_g1 = risk[185]\n",
    "d28_g2 = risk[186]\n",
    "d28_g3 = risk[187]\n",
    "d28_g4 = risk[188]\n",
    "d28_g5 = max(risk[189], risk[190])\n",
    "d28_g6 = risk[191]\n",
    "\n",
    "\n",
    "d28_ratio = (d28_g1*d28_g2*d28_g3*d28_g4*d28_g5)\n",
    "d28_ratio = format(d28_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "###################\n",
    "# D32: 子宮內膜異位症\n",
    "\n",
    "\n",
    "d32 = gene_risk.iloc[192:201]\n",
    "\n",
    "\n",
    "d32_g1 = risk[192]\n",
    "d32_g2 = max(risk[193], risk[194], risk[195])\n",
    "d32_g3 = risk[196]\n",
    "d32_g4 = risk[197]\n",
    "d32_g5 = risk[198]\n",
    "d32_g6 = risk[199]\n",
    "d32_g7 = risk[200]\n",
    "\n",
    "\n",
    "d32_ratio = (d32_g1*d32_g2*d32_g3*d32_g4*d32_g5*d32_g6*d32_g7)\n",
    "d32_ratio = format(d32_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# D33: 子宮肌瘤\n",
    "\n",
    "\n",
    "d33 = gene_risk.iloc[201:207]\n",
    "\n",
    "\n",
    "d33_g1 = max(risk[201], risk[206])\n",
    "d33_g2 = risk[202]\n",
    "d33_g3 = risk[203]\n",
    "d33_g4 = max(risk[204], risk[205])\n",
    "\n",
    "\n",
    "d33_ratio = (d33_g1*d33_g2*d33_g3*d33_g4)\n",
    "d33_ratio = format(d33_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "####################\n",
    "# D34: 多囊性卵巢症候群\n",
    "\n",
    "\n",
    "d34 = gene_risk.iloc[207:217]\n",
    "\n",
    "\n",
    "d34_g1 = max(risk[207], risk[208], risk[209], risk[215], risk[216])\n",
    "d34_g2 = risk[210]\n",
    "d34_g3 = risk[211]\n",
    "d34_g4 = risk[212]\n",
    "d34_g5 = max(risk[213], risk[214])\n",
    "\n",
    "\n",
    "d34_ratio = (d34_g1*d34_g2*d34_g3*d34_g4*d34_g5)\n",
    "d34_ratio = format(d34_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D35: 憂鬱症\n",
    "\n",
    "\n",
    "d35 = gene_risk.iloc[217:224]\n",
    "\n",
    "\n",
    "d35_g1 = risk[217]\n",
    "d35_g2 = risk[218]\n",
    "d35_g3 = max(risk[219], risk[220])\n",
    "d35_g4 = risk[221]\n",
    "d35_g5 = risk[222]\n",
    "d35_g6 = risk[223]\n",
    "\n",
    "\n",
    "d35_ratio = (d35_g1*d35_g2*d35_g3*d35_g4*d35_g5*d35_g6)\n",
    "d35_ratio = format(d35_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "############\n",
    "# D36: 躁鬱症\n",
    "\n",
    "\n",
    "d36 = gene_risk.iloc[224:236]\n",
    "\n",
    "\n",
    "d36_g1 = max(risk[224], risk[225], risk[229], risk[230], risk[233])\n",
    "d36_g2 = max(risk[226], risk[227], risk[228], risk[232])\n",
    "d36_g3 = risk[231]\n",
    "d36_g4 = max(risk[234], risk[235])\n",
    "\n",
    "\n",
    "d36_ratio = (d36_g1*d36_g2*d36_g3*d36_g4)\n",
    "d36_ratio = format(d36_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "################\n",
    "# D37: 思覺失調症\n",
    "\n",
    "\n",
    "d37 = gene_risk.iloc[236:243]\n",
    "\n",
    "\n",
    "d36_g1 = risk[236]\n",
    "d36_g2 = risk[237]\n",
    "d36_g3 = risk[238]\n",
    "d36_g4 = risk[239]\n",
    "d36_g5 = risk[240]\n",
    "d36_g6 = risk[241]\n",
    "\n",
    "\n",
    "d36_ratio = (d36_g1*d36_g2*d36_g3*d36_g4*d36_g5*d36_g6)\n",
    "d36_ratio = format(d36_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "###########\n",
    "# D40: 氣喘\n",
    "\n",
    "\n",
    "d40 = gene_risk.iloc[243:250]\n",
    "\n",
    "\n",
    "d40_g1 = max(risk[243], risk[245])\n",
    "d40_g2 = max(risk[244], risk[246], risk[247])\n",
    "d40_g3 = risk[248]\n",
    "d40_g4 = risk[249]\n",
    "\n",
    "\n",
    "d40_ratio = (d40_g1*d40_g2*d40_g3*d40_g4)*1.1\n",
    "d40_ratio = format(d40_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###############\n",
    "# D41: 攝護腺肥大\n",
    "\n",
    "\n",
    "d41 = gene_risk.iloc[250:256]\n",
    "\n",
    "\n",
    "d41_g1 = risk[250]\n",
    "d41_g2 = risk[251]\n",
    "d41_g3 = risk[252]\n",
    "d41_g4 = max(risk[253], risk[254])\n",
    "d41_g5 = risk[255]\n",
    "\n",
    "\n",
    "d41_ratio = (d41_g1*d41_g2*d41_g3*d41_g4*d41_g5)\n",
    "d41_ratio = format(d41_ratio, '.2f')\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.49\n"
     ]
    }
   ],
   "source": [
    "#############\n",
    "# D45: 膀胱癌\n",
    "\n",
    "\n",
    "d41 = gene_risk.iloc[250:256]\n",
    "\n",
    "\n",
    "d41_g1 = max(risk[256], risk[257], risk[259], risk[265])\n",
    "d41_g2 = risk[251]\n",
    "d41_g3 = risk[252]\n",
    "d41_g4 = max(risk[253], risk[254])\n",
    "d41_g5 = risk[255]\n",
    "\n",
    "\n",
    "d41_ratio = (d41_g1*d41_g2*d41_g3*d41_g4*d41_g5)\n",
    "d41_ratio = format(d41_ratio, '.2f')\n",
    "\n",
    "print(d41_ratio)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Disease_code</th>\n",
       "      <th>rs_number</th>\n",
       "      <th>genotype</th>\n",
       "      <th>risk_encode</th>\n",
       "      <th>risk</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>D41</td>\n",
       "      <td>rs103294</td>\n",
       "      <td>CC</td>\n",
       "      <td>D41rs103294CC</td>\n",
       "      <td>1.4678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>D41</td>\n",
       "      <td>rs339331</td>\n",
       "      <td>CT</td>\n",
       "      <td>D41rs339331CT</td>\n",
       "      <td>0.9328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>D41</td>\n",
       "      <td>rs12621278</td>\n",
       "      <td>AG</td>\n",
       "      <td>D41rs12621278AG</td>\n",
       "      <td>1.0925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>D41</td>\n",
       "      <td>rs743572</td>\n",
       "      <td>AG</td>\n",
       "      <td>D41rs743572AG</td>\n",
       "      <td>0.9128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>D41</td>\n",
       "      <td>rs1800470</td>\n",
       "      <td>AG</td>\n",
       "      <td>D41rs1800470AG</td>\n",
       "      <td>0.9419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>D41</td>\n",
       "      <td>rs1544410</td>\n",
       "      <td>CC</td>\n",
       "      <td>D41rs1544410CC</td>\n",
       "      <td>1.0549</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Disease_code   rs_number genotype      risk_encode    risk\n",
       "250          D41    rs103294       CC    D41rs103294CC  1.4678\n",
       "251          D41    rs339331       CT    D41rs339331CT  0.9328\n",
       "252          D41  rs12621278       AG  D41rs12621278AG  1.0925\n",
       "253          D41    rs743572       AG    D41rs743572AG  0.9128\n",
       "254          D41   rs1800470       AG   D41rs1800470AG  0.9419\n",
       "255          D41   rs1544410       CC   D41rs1544410CC  1.0549"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_risk.iloc[250:256]\n",
    "\n",
    "#print(d25_g1,d25_g2, d25_g3, d25_g4, d25_g5, d25_g6)\n",
    "\n",
    "# -2  -1\n",
    "# -3  -2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 2: Get Subject Data from Google Spreadsheet\n",
    "**會員編號, 檢體編號, 身高, 體重**\n",
    "\n",
    "reference: https://www.twilio.com/blog/2017/02/an-easy-way-to-read-and-write-to-a-google-spreadsheet-in-python.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import gspread\n",
    "from oauth2client.service_account import ServiceAccountCredentials\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "ename": "JSONDecodeError",
     "evalue": "Expecting value: line 1 column 1 (char 0)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mJSONDecodeError\u001b[0m                           Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-28-b95a5de668db>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mcreds\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mServiceAccountCredentials\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_json_keyfile_name\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'client.json'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscope\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/oauth2client/service_account.py\u001b[0m in \u001b[0;36mfrom_json_keyfile_name\u001b[0;34m(cls, filename, scopes, token_uri, revoke_uri)\u001b[0m\n\u001b[1;32m    218\u001b[0m         \"\"\"\n\u001b[1;32m    219\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'r'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mfile_obj\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 220\u001b[0;31m             \u001b[0mclient_credentials\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjson\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile_obj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    221\u001b[0m         return cls._from_parsed_json_keyfile(client_credentials, scopes,\n\u001b[1;32m    222\u001b[0m                                              \u001b[0mtoken_uri\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtoken_uri\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/json/__init__.py\u001b[0m in \u001b[0;36mload\u001b[0;34m(fp, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)\u001b[0m\n\u001b[1;32m    297\u001b[0m         \u001b[0mcls\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcls\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mobject_hook\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobject_hook\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    298\u001b[0m         \u001b[0mparse_float\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparse_float\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparse_int\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparse_int\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 299\u001b[0;31m         parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)\n\u001b[0m\u001b[1;32m    300\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    301\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/json/__init__.py\u001b[0m in \u001b[0;36mloads\u001b[0;34m(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)\u001b[0m\n\u001b[1;32m    352\u001b[0m             \u001b[0mparse_int\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mparse_float\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32mand\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    353\u001b[0m             parse_constant is None and object_pairs_hook is None and not kw):\n\u001b[0;32m--> 354\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_default_decoder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdecode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    355\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mcls\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    356\u001b[0m         \u001b[0mcls\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mJSONDecoder\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/json/decoder.py\u001b[0m in \u001b[0;36mdecode\u001b[0;34m(self, s, _w)\u001b[0m\n\u001b[1;32m    337\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    338\u001b[0m         \"\"\"\n\u001b[0;32m--> 339\u001b[0;31m         \u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mend\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraw_decode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0midx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0m_w\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    340\u001b[0m         \u001b[0mend\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_w\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mend\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mend\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/json/decoder.py\u001b[0m in \u001b[0;36mraw_decode\u001b[0;34m(self, s, idx)\u001b[0m\n\u001b[1;32m    355\u001b[0m             \u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mend\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscan_once\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0midx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    356\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 357\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mJSONDecodeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Expecting value\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    358\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mend\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mJSONDecodeError\u001b[0m: Expecting value: line 1 column 1 (char 0)"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "scope = ['https://docs.google.com/spreadsheets/d/1CPeDwKgeFYBBkLdP_1xBFavplHq-szjGnNlLrs1Bewk/edit#gid=1864317221']\n",
    "#r = requests.get(scope)\n",
    "\n",
    "\n",
    "creds = ServiceAccountCredentials.from_json_keyfile_name('client.json', scope)\n",
    "\n",
    "\n",
    "\n",
    "#client = gspread.authorize(creds)\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "\n",
    "\n",
    "url = 'http://httpbin.org/status/200'\n",
    "r = requests.get(scope)\n",
    "\n",
    "if 'json' in r.headers.get('Content-Type'):\n",
    "    js = r.json()\n",
    "else:\n",
    "    print('Response content is not in JSON format.')\n",
    "    js = 'spam'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 3: Analysis Genotype and Risk of Disease\n",
    "**1. Genotype: delta VIC & FAM** <br><br>\n",
    "**2. Disease Risk Assessment** <br><br>\n",
    "**3. Advice of High Risk Diseases**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NO.</th>\n",
       "      <th>Gene</th>\n",
       "      <th>rs number</th>\n",
       "      <th>VIC</th>\n",
       "      <th>FAM</th>\n",
       "      <th>allele</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>allele3</th>\n",
       "      <th>standard_1</th>\n",
       "      <th>standard_2</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Value</th>\n",
       "      <th>Error</th>\n",
       "      <th>Delta VIC</th>\n",
       "      <th>Delta FAM</th>\n",
       "      <th>VIC signal</th>\n",
       "      <th>FAM signal</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>MS4A6A</td>\n",
       "      <td>rs610932</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G/T</td>\n",
       "      <td>GG</td>\n",
       "      <td>GT</td>\n",
       "      <td>TT</td>\n",
       "      <td>2.20</td>\n",
       "      <td>0.40</td>\n",
       "      <td>GT</td>\n",
       "      <td>1.400971591</td>\n",
       "      <td>NO</td>\n",
       "      <td>52849.17969</td>\n",
       "      <td>37723.23438</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>GAB2</td>\n",
       "      <td>rs7101429</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.30</td>\n",
       "      <td>AA</td>\n",
       "      <td>2.212615574</td>\n",
       "      <td>NO</td>\n",
       "      <td>74289.81250</td>\n",
       "      <td>33575.56250</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>MAPT</td>\n",
       "      <td>rs242562</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.40</td>\n",
       "      <td>AG</td>\n",
       "      <td>0.809437867</td>\n",
       "      <td>NO</td>\n",
       "      <td>118449.98830</td>\n",
       "      <td>146336.10940</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>GAK</td>\n",
       "      <td>rs11248051</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>CC</td>\n",
       "      <td>1.647762233</td>\n",
       "      <td>NO</td>\n",
       "      <td>145549.51560</td>\n",
       "      <td>88331.62500</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>rs3129882</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>1.80</td>\n",
       "      <td>0.50</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.217852979</td>\n",
       "      <td>NO</td>\n",
       "      <td>16721.29688</td>\n",
       "      <td>76754.96094</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   NO.     Gene   rs number VIC FAM allele allele1 allele2 allele3  \\\n",
       "0    1   MS4A6A    rs610932   G   T    G/T      GG      GT      TT   \n",
       "1    2     GAB2   rs7101429   A   G    A/G      AA      AG      GG   \n",
       "2    3     MAPT    rs242562   A   G    A/G      AA      AG      GG   \n",
       "3    4      GAK  rs11248051   C   T    C/T      CC      CT      TT   \n",
       "4    5  HLA-DRA   rs3129882   A   G    A/G      AA      AG      GG   \n",
       "\n",
       "   standard_1  standard_2 Genotype        Value Error     Delta VIC  \\\n",
       "0        2.20        0.40       GT  1.400971591    NO   52849.17969   \n",
       "1        0.85        0.30       AA  2.212615574    NO   74289.81250   \n",
       "2        0.85        0.40       AG  0.809437867    NO  118449.98830   \n",
       "3        1.00        0.01       CC  1.647762233    NO  145549.51560   \n",
       "4        1.80        0.50       GG  0.217852979    NO   16721.29688   \n",
       "\n",
       "      Delta FAM  VIC signal  FAM signal  \n",
       "0   37723.23438           1           1  \n",
       "1   33575.56250           1           1  \n",
       "2  146336.10940           1           1  \n",
       "3   88331.62500           1           1  \n",
       "4   76754.96094           1           1  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#############################\n",
    "#1. Genotype: delta VIC & FAM \n",
    "#############################\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "genotype = pd.read_csv(open('./raw/genotype.csv', 'rb'))\n",
    "     \n",
    "\n",
    "genotype.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "i like most important thing is that I need to be with you. I nned to play with you, I need a girl to take care of me. I want to hug you and \n",
    "\n",
    "and put you un my arm because i like you. you are the one i want and you are the one i love."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
